docs/manual/programmability/lua-api.md

```diff
@@ -454,25 +454,29 @@ You can use the following flags and instruct the server to treat the scripts' ex
 
 * `no-writes`: this flag indicates that the script only reads data but never writes.
 
-    By default, Redis will deny the execution of scripts against read-only replicas, as they may attempt to perform writes.
+    By default, Redis will deny the execution of flagged scripts (Functions and Eval scripts with [shebang](/topics/eval-intro#eval-flags)) against read-only replicas, as they may attempt to perform writes.
     Similarly, the server will not allow calling scripts with `FCALL_RO` / `EVAL_RO`.
     Lastly, when data persistence is at risk due to a disk error, execution is blocked as well.
 
     Using this flag allows executing the script:
-    1. With `FCALL_RO` / `EVAL_RO` against masters and read-only replicas.
-    2. Even if there's a disk error (Redis is unable to persist so it rejects writes).
+    1. With `FCALL_RO` / `EVAL_RO`
+    2. On read-only replicas.
+    3. Even if there's a disk error (Redis is unable to persist so it rejects writes).
+    4. When over the memory limit since it implies the script doesn't increase memory consumption (see `allow-oom` below)
 
     However, note that the server will return an error if the script attempts to call a write command.
+    Also note that currently `PUBLISH`, `SPUBLISH` and `PFCOUNT` are also considered write commands in scripts, because they could attempt to propagate commands to replicas and AOF file.
```

Comment:
Should we mentioned that its on 7.0.1 and above?

Reply:
i don't know.. this whole section is about a new feature of 7.0 (flags). so if we document that, then it's about a behavior change between 7.0.0 and 7.0.1.
but then, maybe we should document that in despite what the text says, in 7.0.0 `no-writes` did not imply `allow-oom`, and that you had to add it explicitly?
i think this could cause a lot of confusion (the text will be hard to follow).

maybe i'll just add some `history` section and keep it separate from the rest of the text.

Reply:
actually, i now recall that #1953 adds a new section specific to read-only scripts (currently located in the wrong file IMHO).
i guess we should squash these two PRs together, or merge the other and then modify the text again here.